vibrato/src/dictionary/connector/dual_connector.rs

```diff
@@ -0,0 +1,296 @@
+use std::io::Read;
+
+use bincode::{Decode, Encode};
+use hashbrown::{HashMap, HashSet};
+
+use crate::dictionary::connector::raw_connector::scorer::{
+    Scorer, ScorerBuilder, U31x8, SIMD_SIZE,
+};
+use crate::dictionary::connector::raw_connector::{RawConnectorBuilder, INVALID_FEATURE_ID};
+use crate::dictionary::connector::{Connector, ConnectorCost, MatrixConnector};
+use crate::dictionary::mapper::ConnIdMapper;
+use crate::errors::Result;
+use crate::num::U31;
+
+#[derive(Decode, Encode)]
+pub struct DualConnector {
+    matrix_connector: MatrixConnector,
+    matrix_left_id_map: Vec<u16>,
+    matrix_right_id_map: Vec<u16>,
+    raw_right_ids: Vec<U31x8>,
+    raw_left_ids: Vec<U31x8>,
+    raw_scorer: Scorer,
+}
+
+impl DualConnector {
+    /// Removes feature templates so that the matrix size is smaller using greedy search
+    /// and returns a set of rest IDs.
+    pub fn remove_feature_templates_greedy(
+        raw_id_size: usize,
```

Comment:
```suggestion
        raw_idx_size: usize,
```
Could you use `idx` for consistency?

Reply:
I changed the name to `raw_feat_template_size`.